mlrun/k8s_utils.py

```diff
@@ -475,6 +475,37 @@ def get_project_secret_data(self, project, secret_keys=None, namespace=""):
 
         return results
 
+    def list_services(self, namespace=None, states=None):
+        try:
+            services = self.v1api.list_namespaced_service(
+                self.resolve_namespace(namespace)
+            )
+        except ApiException as exc:
+            logger.error(f"failed to list services: {exc}")
+            raise exc
+
+        return [
+            service
+            for service in services.items
+            if not states or service.status.phase in states
+        ]
+
+    def is_service_exist(self, name, namespace=None, is_release_name: bool = True):
+        items = self.list_services(namespace=namespace)
+        for item in items:
+            if (
+                is_release_name
+                and item.metadata.annotations is not None
+                and "meta.helm.sh/release-name" in item.metadata.annotations
```

Comment:
Verify with @Hedingber its a valid / robust check. Maybe add a configuration from `mlconf` if its not something set in stone.